your_visibility/starter_dacx.py

```diff
@@ -0,0 +1,64 @@
+import asyncio
+
+from temporalio.client import Client
+
+from workflow_dacx import GreetingWorkflow
+
+"""dacx
+To set custom Search Attributes, use the `search_attributes` parameter of the ['start_workflow()'](https://python.temporal.io/temporalio.client.Client.html#start_workflow) method.
+dacx"""
+
+"""dacx
+Use the [list_workflows()](https://python.temporal.io/temporalio.client.Client.html#list_workflows) method on the Client handle and pass a [List Filter](/concepts/what-is-a-list-filter) as an argument to filter the listed Workflows.
+dacx"""
+
+
+async def main():
+    client = await Client.connect("localhost:7233")
+    handle = await client.start_workflow(
+        GreetingWorkflow.run,
+        id="search-attributes-workflow-id",
+        task_queue="search-attributes-task-queue",
+        # Start with default set of search attributes
+        search_attributes={"CustomKeywordField": ["old-value"]},
+    )
+
+    print(
+        "First search attribute values: ",
+        (await handle.describe()).search_attributes.get("CustomKeywordField"),
+    )
+    await asyncio.sleep(3)
+    print(
+        "Second search attribute values: ",
+        (await handle.describe()).search_attributes.get("CustomKeywordField"),
+    )
+    await asyncio.sleep(3)
+    print(
+        "Empty search attribute values: ",
+        (await handle.describe()).search_attributes.get("CustomKeywordField"),
+    )
+
+    async for workflow in client.list_workflows('WorkflowType="GreetingWorkflow"'):
+        print(f"Workflow: {workflow.id}")
+    await handle.cancel()
```

Comment:
Does this succeed? I would expect workflow to be complete by this time

Reply:
Ah, I confirmed we don't fail a cancel if workflow is already complete. So this is a no-op and should just be removed (workflow is already closed by this time, nothing to cancel).